### PR TITLE
Allow logged-out users to view pages

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

This PR makes it so that if you visit a page's URL while you are logged out, you will see a readonly version of the page, stripped of any UI.

This involves removing the 'is logged in' middleware on the API `page` query, but as per the README the app is not yet secure and this middleware was only ever superficial (it is possible to retrieve entities you do not own via other routes, and there is self sign-up, making the app as a whole insecure). A proper 'view page while logged out' solution would involve users either (a) choosing to make particular pages public or private, or (b) generating 'anyone with the link can view' links which encode a token of some kind.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1203216920361670/f) _(internal)_

## 🔍 What does this change?

- Allow anonymous retrieval of pages
- Update `useReadonlyMode` to also put the app into readonly if there is no authenticated user
- Also hide the navbar and comment UI if in readonly mode
- don't load ProseMirror plugins if in readonly (they're all related to editing)

## ⚠️ Known issues

- This is a temporary solution that makes it even easier for anyone to see any page.
- The page content doesn't seem quite centered to me, but this also happens in editing view. We should fix that at some point

## 🐾 Next steps

- A better solution when we introduce proper auth

## ❓ How to test this?

1.  Get a page URL by logging in and visiting a page
2. Open the page URL in incognito
3. See the page in readonly mode

## 📹 Demo

Screenshot 2022-11-03 at 17 56 45

### CODE REVIEW

File: packages/hash/frontend/src/blocks/page/PageBlock.tsx [hunk 1]
```diff
@@ -146,13 +146,14 @@ export const PageBlock: FunctionComponent<PageBlockProps> = ({
...
                 }}
               >
-                {pageComments?.map((comment) => (
-                  <CommentThread
-                    key={comment.entityId}
-                    pageId={entityId}
-                    comment={comment}
-                  />
-                ))}
+                {!readonlyMode &&
```
Comment (luisbettencourt) [inline packages/hash/frontend/src/blocks/page/PageBlock.tsx:149]: I think this should be above:

```
{readonlyMode || loadingTypeSystem ? null : (
          <PageSectionContainer 
...
```

Comment (luisbettencourt) [inline packages/hash/frontend/src/blocks/page/PageBlock.tsx:149]: Comments slightly change the styling of the PageSectionContainer component, so if we're hiding them in readOnly mode we should also prevent that change from happening:

` const commentsContainerWidth = !readonlyMode && pageComments?.length ? COMMENTS_WIDTH : 0;`

**Review by luisbettencourt — CHANGES_REQUESTED** [review]

**Review by yusufkinatas — COMMENTED** [review]
I agree with @luisbettencourt 's comments. 
Tested the branch on my local, it works as expected. Looks good to me.

I guess we would want to add the top nav bar back to the page at some point right? Otherwise there is no way to "create a user" when looking at a page as a guest.

File: packages/hash/frontend/src/blocks/page/PageSectionContainer.tsx [hunk 2]
```diff
@@ -10,9 +11,9 @@ export const getPageSectionContainerStyles = (pageComments: PageThread[]) => {
   const commentsContainerWidth = pageComments?.length ? COMMENTS_WIDTH : 0;
 
   const paddingLeft = `max(calc((100% - ${
-    PAGE_CONTENT_WIDTH + commentsContainerWidth + PAGE_MIN_PADDING
+    PAGE_CONTENT_WIDTH + commentsContainerWidth
```
Comment (CiaranMn) [inline packages/hash/frontend/src/blocks/page/PageSectionContainer.tsx:14]: @luisbettencourt this was adding additional padding to the left, was this intentional?

before
Screenshot 2022-11-04 at 09 20 18


after
Screenshot 2022-11-04 at 09 20 02

File: packages/hash/frontend/src/blocks/page/PageBlock.tsx (hunk 1 above)
Comment (CiaranMn) [reply packages/hash/frontend/src/blocks/page/PageBlock.tsx]: Thank you, done in https://github.com/hashintel/hash/pull/1340/commits/8bb13cef78aee51859dac9347881d8f4f79268dd

Comment (CiaranMn) [reply packages/hash/frontend/src/blocks/page/PageBlock.tsx]: Done in https://github.com/hashintel/hash/pull/1340/commits/8bb13cef78aee51859dac9347881d8f4f79268dd

**Comment by CiaranMn** [comment]
> I agree with @luisbettencourt 's comments. Tested the branch on my local, it works as expected. Looks good to me.
> 
> I guess we would want to add the top nav bar back to the page at some point right? Otherwise there is no way to "create a user" when looking at a page as a guest.

@yusufkinatas yes agreed, we will want to do that – and replace this implementation entirely

File: packages/hash/frontend/src/blocks/page/PageSectionContainer.tsx [hunk 3]
```diff
@@ -10,9 +11,9 @@ export const getPageSectionContainerStyles = (pageComments: PageThread[]) => {
   const commentsContainerWidth = pageComments?.length ? COMMENTS_WIDTH : 0;
```
Comment (luisbettencourt) [inline packages/hash/frontend/src/blocks/page/PageSectionContainer.tsx:11]:
```suggestion
  const commentsContainerWidth = pageComments?.length ? COMMENTS_WIDTH + PAGE_MIN_PADDING : 0;
```

**Review by luisbettencourt — CHANGES_REQUESTED** [review]

File: packages/hash/frontend/src/blocks/page/PageSectionContainer.tsx (hunk 2 above)
Comment (luisbettencourt) [reply packages/hash/frontend/src/blocks/page/PageSectionContainer.tsx:14]: Nice spot @CiaranMn !

It wasn't intentional, it was only supposed to be applied when there are comments in the page.
I've made a suggestion here: https://github.com/hashintel/hash/pull/1340#discussion_r1013944672

Comment (CiaranMn) [reply packages/hash/frontend/src/blocks/page/PageSectionContainer.tsx:14]: Understood, and thank you for the suggestion, applied!

**Review by luisbettencourt — CHANGES_REQUESTED** [review]
Sorry for drip-feeding requested changes, but I just noticed that when logging in the sidebar and nav bar are hidden until the page is reloaded

File: packages/hash/frontend/src/pages/[account-slug]/index.page.tsx [hunk 4]
```diff
@@ -8,10 +8,14 @@ import { Link } from "../../shared/ui";
...
 
 const Page: NextPageWithLayout = () => {
-  const { authenticatedUser } = useAuthenticatedUser();
+  const { loading, authenticatedUser } = useAuthenticatedUser();
   const { orgs } = useOrgs();
   const { accountId } = useRouteAccountInfo();
 
+  if (loading) {
+    return null;
+  }
```
Comment (CiaranMn) [inline packages/hash/frontend/src/pages/[account-slug]/index.page.tsx:17]: This was previously flashing up the "You must be logged in..." message when the session was loading.

**Review by luisbettencourt — APPROVED** [review]
Thank you for bearing with the changes I requested

**Review by luisbettencourt — APPROVED** [review]

**Pull request merged**